### DISCUSSION TITLE
Wave 3.x Upgrade Features

### DISCUSSION THREAD

**Discussion by tnylea** [thedevdojo/wave#112] [General]
The purpose for this discussion is to highlight some of the features/improvements that will be added to Wave V3.

- [x] Upgrade to Laravel 11
- [x] Upgrade to Livewire V3
- [x] Replace Roles/Permissions with Spatie Permissions
- [x] Remove Voyager
- [x] Add Filament
- [ ] (Possible) Add Wave as a separate package that can be installed via Composer
- [ ] Teams Support
- [x] Upgraded Billing Support to include Latest Paddle, Stripe, and possible LemonSqueezy
- [x] Add Resources to Filament admin similar to Voyager
- [x] Default to the Tallstack theme (allows for full livewire/TALL functionality)
- [x] Add default vite config with Asset builder at the root of the folder. No more node_modules in individual theme folders.
- [ ] Confirm all existing tests pass for the new version and possibly add more tests
- [x] Much more

If there is anything specific you would like to see in the new release, be sure to post your comments/ideas below.

Thanks!

**Comment by larson-carter** [comment]
@tnylea I was just about to cross reference this discussion on this one [here](https://github.com/thedevdojo/voyager/issues/5846).

Since you are removing voyager what are you going todo instead? 

I do see that Laravel 11 support is enabled which is amazing!!! :)

**Reply by bobbyiliev** [reply]
> Since you are removing voyager what are you going todo instead?

We are planning to use Filament instead!

**Reply by larson-carter** [reply]
Gotcha, that sounds great. I just found this project I will def try it out.

**Comment by MrGKanev** [comment]
1. Is Wave 3.X open for PRs that aren't connected with the prior mentioned features? - SEO changes for example? 
2. What is the ETA for this version?

**Reply by bobbyiliev** [reply]
The v3.x is in a very early stage, first work was done just this weekend so we don't have an ETA yet.

PRs are welcome as well as any suggestions! Though for PRs it might be a bit dynamic initially while the groundwork is being done.

**Comment by KeithPrinkey-ops** [comment]
> The purpose for this discussion is to highlight some of the features/improvements that will be added to Wave V3.
> 
> * [x]  Upgrade to Laravel 11
> * [x]  Upgrade to Livewire V3
> * [x]  Replace Roles/Permissions with Spatie Permissions
> * [x]  Remove Voyager
> * [x]  Add Filament
> * [x]  (Possible) Add Wave as a separate package that can be installed via Composer
> * [x]  Teams Support
> * [x]  Upgraded Billing Support to include Latest Paddle, Stripe, and possible LemonSqueezy
> * [x]  Add Resources to Filament admin similar to Voyager
> * [x]  Default to the Tallstack theme (allows for full livewire/TALL functionality)
> * [x]  Add default vite config with Asset builder at the root of the folder. No more node_modules in individual theme folders.
> * [x]  Confirm all existing tests pass for the new version and possibly add more tests
> * [ ]  Much more
> 
> If there is anything specific you would like to see in the new release, be sure to post your comments/ideas below.
> 
> Thanks!

I think you guys already know my stance on Wave V3. Couldn't be more excited. Makes the devs job so much easier. I assume API will be standard out of the box again? Is this a full rebuild or an upgrade of V2? I am just curious if I will be able to upgrade from V2 is all. 

I see filament is going to be used. Is it possible to have the admin panel as a separate package so people can chose which panel they prefer? Personally, I prefer Voyager. However, you customize one admin panel, you can customize them all. Preferring Voyager is an opinion really. I have not tried Filament yet, is it better than Voyager? I though Dev Dojo managed Voyager now? Any particular reason Voyager was passed on? Sorry man, I am just curious. 

If you are considering Teams integration, I would highly recommend JetStream/Fortify unless the plan is to make it custom to Wave V3 but even then you could integrate JetStream.

**Comment by l-alexandrov** [comment]
Will we be able to upgrade from 2.2.0 along with using the newest features introduced in Wave 3?

 **_P.S. I think packaging Wave making it `composer`-installable is more than beneficial for the project._**

**Comment by mikemastercorp** [comment]
Hello guys. Thanks for the update on v3. I am also very excited and can't wait to get my hands around a combination of Wave + filament which I adore and have used for many projects now.

I don't think voyager is bad (don't get me wrong but what I can do with filament and it's full ability with Livewire 3 + AlpuneJS is amazing and I am sure this will lead to a much bigger interest of the final version from other devs being still just Filament Devs.

I would.allow to second the option to upgrade Wave from one to the other version. (v2 to V3) As many of us already have some projects with Wave 2 and we can all benefit from the upgrade.

One other thing I would suggest is to focus a bit more on theming the admin(allowing custom templates to be implemented).  I know filament already supports theming which can be extended further allowing everyone to grab a custom/preferable admin and use it's templates.to adapt the views etc.

Maybe one other thing can help too - modularity of features (like Laravel Modules or else) which would allow the community to build around some module/plugins to be proud and share/sell even depending on the amount of work etc.

And lastly, but not the least - update the video gallery as to be honest this was one weak point of.Wave SAAS as I was enjoying the learning curve but from some point there weren't many more new videos.

Well If.I get.some.more ideas, I will definitely share and.would.likw to take the chance to thank you for all the efforts on behalf of all the happy wave.saas community users!

**Comment by iambhavikmistry** [comment]
@tnylea , would you be able to share an estimated timeline for Wave 3.0?

**Reply by iambhavikmistry** [reply]
@tnylea @bobbyiliev would you be able to share an estimated timeline for Wave 3.0?

**Reply by bobbyiliev** [reply]
Probably next month or so! We don’t have an exact ETA though.

**Reply by cjdupreez** [reply]
Hey @bobbyiliev, just checking in on this to see if you've got an ETA yet for the 3.0 release? Looking forward to it!

**Reply by iambhavikmistry** [reply]
@cjdupreez it's released.

**Reply by cjdupreez** [reply]
Lol, whelp... that's what I get for not refreshing. Thanks!

**Reply by bobbyiliev** [reply]
Ah yes, sorry forgot to update this discussion! 

Let me know how it goes if you give v3 a try!

**Comment by jessijean** [comment]
Hi everyone, Wave seems like an awesome framework for my Laravel project. Thanks for all your hard work! I saw on this thread you were thinking of adding lemonsqueezy. Is this something you are still considering? I'd love to take that approach over stripe.

**Reply by bobbyiliev** [reply]
Hey 👋 

As of the time being we are not actively working on adding support for LemonSqueezy officially. However, if this is a must for you, you should be able to get that working with some customizations and possibly using the official Laravel package that they now offer:

> https://www.lemonsqueezy.com/blog/laravel-lemon-squeezy

Let me know if you hit any issues while integrating it!
